tests/components/rfxtrx/test_cover.py

```diff
@@ -47,151 +29,148 @@ def test_valid_config(self):
-        assert 1 == len(rfxtrx_core.RFX_DEVICES)
-
-        event = rfxtrx_core.get_rfx_object("0a1400adf394ab020e0060")
-        event.data = bytearray(
-            [0x0A, 0x14, 0x00, 0xAD, 0xF3, 0x94, 0xAB, 0x02, 0x0E, 0x00, 0x60]
-        )
-
-        for evt_sub in rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS:
-            evt_sub(event)
-        assert 2 == len(rfxtrx_core.RFX_DEVICES)
-
-        # Trying to add a sensor
-        event = rfxtrx_core.get_rfx_object("0a52085e070100b31b0279")
-        event.data = bytearray(b"\nR\x08^\x07\x01\x00\xb3\x1b\x02y")
-        for evt_sub in rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS:
-            evt_sub(event)
-        assert 2 == len(rfxtrx_core.RFX_DEVICES)
-
-        # Trying to add a light
-        event = rfxtrx_core.get_rfx_object("0b1100100118cdea02010f70")
-        event.data = bytearray(
-            [0x0B, 0x11, 0x11, 0x10, 0x01, 0x18, 0xCD, 0xEA, 0x01, 0x02, 0x0F, 0x70]
-        )
-        for evt_sub in rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS:
-            evt_sub(event)
-        assert 2 == len(rfxtrx_core.RFX_DEVICES)
-
-    def test_discover_cover_noautoadd(self):
-        """Test with discovery of cover when auto add is False."""
-        assert setup_component(
-            self.hass,
-            "cover",
-            {"cover": {"platform": "rfxtrx", "automatic_add": False, "devices": {}}},
-        )
-
-        event = rfxtrx_core.get_rfx_object("0a1400adf394ab010d0060")
-        event.data = bytearray(
-            [0x0A, 0x14, 0x00, 0xAD, 0xF3, 0x94, 0xAB, 0x01, 0x0D, 0x00, 0x60]
-        )
-
-        for evt_sub in rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS:
-            evt_sub(event)
-        assert 0 == len(rfxtrx_core.RFX_DEVICES)
 
-        event = rfxtrx_core.get_rfx_object("0a1400adf394ab020e0060")
-        event.data = bytearray(
-            [0x0A, 0x14, 0x00, 0xAD, 0xF3, 0x94, 0xAB, 0x02, 0x0E, 0x00, 0x60]
-        )
-        for evt_sub in rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS:
-            evt_sub(event)
-        assert 0 == len(rfxtrx_core.RFX_DEVICES)
-
-        # Trying to add a sensor
-        event = rfxtrx_core.get_rfx_object("0a52085e070100b31b0279")
-        event.data = bytearray(b"\nR\x08^\x07\x01\x00\xb3\x1b\x02y")
-        for evt_sub in rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS:
-            evt_sub(event)
-        assert 0 == len(rfxtrx_core.RFX_DEVICES)
-
-        # Trying to add a light
-        event = rfxtrx_core.get_rfx_object("0b1100100118cdea02010f70")
-        event.data = bytearray(
-            [0x0B, 0x11, 0x11, 0x10, 0x01, 0x18, 0xCD, 0xEA, 0x01, 0x02, 0x0F, 0x70]
-        )
-        for evt_sub in rfxtrx_core.RECEIVED_EVT_SUBSCRIBERS:
-            evt_sub(event)
-        assert 0 == len(rfxtrx_core.RFX_DEVICES)
+async def test_default_config(hass, rfxtrx):
+    """Test with 0 cover."""
+    assert await async_setup_component(
+        hass, "cover", {"cover": {"platform": "rfxtrx", "devices": {}}}
+    )
+    assert 0 == len(rfxtrx_core.RFX_DEVICES)
```

Comment:
The devices should be moved to hass.data.

Reply:
I think I can do away with it fully with using entity registry. That is my hope anyway.